aggregate_key_discovery.md

```diff
@@ -0,0 +1,90 @@
+    * then, the bucket is checked against the threshold the ad tech declared. The aggregation service will only include buckets in the output whose noisy values exceed that threshold. Note that this means some buckets may be present in the resulting summary report that have no associated real user contributions.
+
+#### Precision/recall tradeoff
+
+To ensure privacy, noise will be added to all requested buckets before thresholding, even if they are not part of the aggregatable reports. Therefore, buckets without any input may have non-zero values and appear similar to buckets that had input. Thus, a threshold will be chosen that minimizes the number of buckets in the output that are pure noise values (high precision) and maximizes the number of buckets with input (high recall).
+
+The adtech-specified threshold works as a lever to control a trade off between the precision and recall of the output. 
+
+![Precision recall tradeoff diagram](precision-recall-tradeoff.svg)
+_Fig 1: The chart shows two overlapping distributions, one in blue for false conversions (no user contribution) and one in orange for true conversions (have user contribution). Two distributions are divided by a threshold value, buckets that exceed the threshold (positive) are reported while the buckets that do not exceed the threshold (negative) are not reported in the output._
+
+Here precision is the proportion of buckets with user contribution in the output, while recall is the proportion of buckets with user contribution that are correctly reported in the output among all buckets with user contribution.
+
+Both higher precision and higher recall are better, but they are inversely related. Adtech can set thresholds for balancing accuracy/recall with each query. By default the threshold is set to a value that provides 100% precision (`L1 + L1/epsilon*ln(1/delta)`) though at that threshold value recall may be low, as the threshold is set to a relatively high value. Assuming the values of epsilon=10, delta=10^-8, and L1=2^16, then the threshold will be 186257 (2.84*L1) and expected recall will be ~82% for a [sample ads dataset](https://ailab.criteo.com/criteo-sponsored-search-conversion-log-dataset).
+
+
+## Threshold selection
+
+We intend to publish a tool which demonstrates the impact of threshold selection for precision/recall tradeoff. With this tool, we intend for adtechs to be able to:
+
+* Explore impact of threshold on various key-mask, epsilon, delta, with simulated datasets.
+* Find the threshold level that satisfies adtech’s expectations.
+* Learn the estimated output size for user-defined settings.
+
+
+## Backward compatible API call
+
+Key-mask by default is all zero (meaning that the default threshold will never kick in) which would emulate the legacy aggregate API call where only buckets in pre_declared_labels are present in the output.
```

Comment:
What does it mean to "kick in"? Maybe "no bucket with true user contribution will be thresholded"?

Reply:
threshold never **kick in** here means threshold would not be applied to any bucket as no key would match all-zero default key-mask. 
to improve readability, I am going to place it with following text. does that sound better?
"The key-mask is all zeros by default, so no extra keys will be discovered and the default threshold will never be triggered. This means that the output will only contain buckets that are in pre_declared_labels, which is similar to the legacy aggregate API call."

Reply:
LGTM!